### PR TITLE
done fix planning screen

### PR DESCRIPTION
**Pull request by ian-bpoc**
# Pull Request Review Comments

## Overview
- **Purpose of the PR**:
    - [ ] Feature addition
    - [/ ] Bug fix
    - [/ ] Refactoring
    - [/ ] Documentation update

## Specific Areas for Verification
- Please specify any particular requests or questions for the reviewer here.

## Suggestions/Concerns
- If you have any suggestions or concerns, please mention them here.

---

_**Additional Comments for the Reviewer (Optional)**_

- Add any additional comments or information for the reviewer here.

### CODE REVIEW

File: client/src/components/Table/TablePlanningA.tsx [hunk 1]
```diff
@@ -59,134 +91,110 @@ const TablePlanning = () => {
...
-        const totalRevenue = aggregatedData[month]?.product_purchases + aggregatedData[month]?.purchases || 0;
-        const grossProfit = totalRevenue - (aggregatedData[month]?.cost_of_sales || 0);
-        return grossProfit;
+      
+      // SELLING AND GENERAL ADMIN EXPENSES
+      const sellingAndGeneralAdminExpenseValues = months.map((month, index) => {
+        const total_employee_expense = totalEmployeeExpenseValues[index]; // Get the sales revenue for the current month
+        const total_expense = totalExpenseValues[index]; // Get the cost of sales for the current month
+        const sellingAndGeneralAdminExpense = total_employee_expense + total_expense; // Calculate gross profit
+        return sellingAndGeneralAdminExpense;
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:177]: Lets fix  the comments so they are correct- i think they were copied from above!

File: client/src/components/Table/TablePlanningA.tsx [hunk 2]
```diff
@@ -59,134 +91,110 @@ const TablePlanning = () => {
...
-      };
-
-      const sellingGeneralValues = generalExpenseValues.map(item => item.generalTotal);
-      const operatingProfitValues = grossProfitValues.map((grossProfit, index) => grossProfit - sellingGeneralValues[index]);
+      // OPERATING INCOME
+      const operatingIncomeValues = months.map((month, index) => {
+        const gross_profit = grossProfitValues[index]; // Get the sales revenue for the current month
+        const selling_and_general_admin = sellingAndGeneralAdminExpenseValues[index]; // Get the cost of sales for the current month
+        const operating_income_value = gross_profit - selling_and_general_admin; // Calculate gross profit
+        return operating_income_value;
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:185]: Lets correct the comments - i think they were copied from above!

File: client/src/components/Table/TablePlanningA.tsx [hunk 3]
```diff
@@ -295,16 +320,23 @@ const TablePlanning = () => {
...
+        {
+          label: 'grossProfit',
+          values: [
+            ...grossProfitValues,
+            firstHalfTotal(grossProfitValues),
+            secondHalfTotal(grossProfitValues),
+            total(grossProfitValues),
+            ''
+          ],
+        },//gross profit
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:332]: do we need this comment? What is it for?

File: client/src/components/Table/TablePlanningA.tsx [hunk 4]
```diff
@@ -295,16 +320,23 @@ const TablePlanning = () => {
...
           values: [
-            ...personnelExpensesValues,
-            firstHalfTotal(personnelExpensesValues),
-            secondHalfTotal(personnelExpensesValues),
-            total(personnelExpensesValues),
+            ...totalEmployeeExpenseValues,
+            firstHalfTotal(totalEmployeeExpenseValues),
+            secondHalfTotal(totalEmployeeExpenseValues),
+            total(totalEmployeeExpenseValues),
             // `${(total(personnelExpensesValues) / total(costOfSalesValues) * 100).toFixed(2)}%`,
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:340]: I think employeeExpenseValues is not related to costOfSales, right? so maybe we don't need this code/comment.

File: client/src/components/Table/TablePlanningA.tsx [hunk 5]
```diff
@@ -323,56 +355,67 @@ const TablePlanning = () => {
...
           values: [
-            ...assign_unit_priceValues,
-            firstHalfTotal(assign_unit_priceValues),
-            secondHalfTotal(assign_unit_priceValues),
-            total(assign_unit_priceValues),
+            ...salaryValues,
+            firstHalfTotal(salaryValues),
+            secondHalfTotal(salaryValues),
+            total(salaryValues),
             // `${(total(consumableValues) / total(costOfSalesValues) * 100).toFixed(2)}%`,
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:362]: should this not be something like this?

            // `${(total(salaryValues) / total(employeeExpensesValues) * 100).toFixed(2)}%`,

File: client/src/components/Table/TablePlanningA.tsx [hunk 6]
```diff
@@ -323,56 +355,67 @@ const TablePlanning = () => {
...
           values: [
-            ...travelExpenseValues,
-            firstHalfTotal(travelExpenseValues),
-            secondHalfTotal(travelExpenseValues),
-            total(travelExpenseValues),
+            ...fuelAllowanceValues,
+            firstHalfTotal(fuelAllowanceValues),
+            secondHalfTotal(fuelAllowanceValues),
+            total(fuelAllowanceValues),
             // `${(total(travelExpenseValues) / total(costOfSalesValues) * 100).toFixed(2)}%`,
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:373]: should this not be something like this?

        // `${(total(fuelAllowanceValues) / total(employeeExpensesValues) * 100).toFixed(2)}%`,

File: client/src/components/Table/TablePlanningA.tsx [hunk 7]
```diff
@@ -323,56 +355,67 @@ const TablePlanning = () => {
...
           values: [
-            ...taxesPublicChargesValues,
-            firstHalfTotal(taxesPublicChargesValues),
-            secondHalfTotal(taxesPublicChargesValues),
-            total(taxesPublicChargesValues),
+            ...statutoryWelfareExpenseValues,
+            firstHalfTotal(statutoryWelfareExpenseValues),
+            secondHalfTotal(statutoryWelfareExpenseValues),
+            total(statutoryWelfareExpenseValues),
             // `${(total(taxesPublicChargesValues) / total(costOfSalesValues) * 100).toFixed(2)}%`,
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:384]: should this not be something like this?

    // `${(total(statutoryWelfareExpenseValues) / total(employeeExpensesValues) * 100).toFixed(2)}%`,
    
    
    PLEASE CHECK ALL COMMENTS LIKE THIS

File: client/src/components/Table/TablePlanningA.tsx [hunk 8]
```diff
@@ -323,56 +355,67 @@ const TablePlanning = () => {
...
           values: [
-            ...utilitiesValues,
-            firstHalfTotal(utilitiesValues),
-            secondHalfTotal(utilitiesValues),
-            total(utilitiesValues),
+            ...welfareExpenseValues,
+            firstHalfTotal(welfareExpenseValues),
+            secondHalfTotal(welfareExpenseValues),
+            total(welfareExpenseValues),
             // `${(total(utilitiesValues) / total(costOfSalesValues) * 100).toFixed(2)}%`,
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:395]: Please Fix Comment values

File: client/src/components/Table/TablePlanningA.tsx [hunk 9]
```diff
@@ -323,56 +355,67 @@ const TablePlanning = () => {
...
           ],
         },
+        {
+          label: 'insurancePremiums',
+          values: [
+            ...insurancePremiumsValues,
+            firstHalfTotal(insurancePremiumsValues),
+            secondHalfTotal(insurancePremiumsValues),
+            total(insurancePremiumsValues),
+            // `${(total(paymentFeeValues) / total(costOfSalesValues) * 100).toFixed(2)}%`,
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:406]: Please Fix Comment values

File: client/src/components/Table/TablePlanningA.tsx [hunk 10]
```diff
@@ -323,56 +355,67 @@ const TablePlanning = () => {
...
+            '0',
+          ],
+        },
         //end for planning portion
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:410]: Is this end of planning protion?
Is it not end of the employee expenses section?

File: client/src/components/Table/TablePlanningA.tsx [hunk 11]
```diff
@@ -59,134 +91,110 @@ const TablePlanning = () => {
...
-      const communicationExpenseValues = months.map(month => aggregatedExpensesData[month]?.communication_expenses || 0);
-      const advertisingExpenseValues = months.map(month => aggregatedExpensesData[month]?.advertising_expenses || 0);
-      const entertainmentExpenseValues = months.map(month => aggregatedExpensesData[month]?.entertainment_expenses || 0);
+      const costOfSalesValues = months.map(month => {
+        const purchases = aggregatedData[month]?.purchase || 0;
+        const outsourcing = aggregatedData[month]?.outsourcing_expense || 0;
+        const productPurchase = aggregatedData[month]?.product_purchase || 0;
+        const dispatchLabor = aggregatedData[month]?.dispatch_labor_expense || 0;
+        const communicationCost = aggregatedData[month]?.communication_expense || 0;
+        const inProgress = aggregatedData[month]?.work_in_progress_expense || 0;
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:105]: lets make this workInProgress not inProgress

File: client/src/components/Table/TablePlanningA.tsx [hunk 12]
```diff
@@ -59,134 +91,110 @@ const TablePlanning = () => {
...
-      const taxesPublicChargesValues = months.map(month => aggregatedExpensesData[month]?.taxes_and_public_charges || 0);
-      const utilitiesValues = months.map(month => aggregatedExpensesData[month]?.utilities_expenses || 0);
-
-      //FOR EXPENSES
-      const rentValues = months.map(month => aggregatedExpensesData[month]?.rent || 0);
-      const paymentFeeValues = months.map(month => aggregatedExpensesData[month]?.payment_fees || 0);
-      const depreciationExpensesValues = months.map(month => aggregatedExpensesData[month]?.depreciation_expenses || 0);
-      const communicationExpenseValues = months.map(month => aggregatedExpensesData[month]?.communication_expenses || 0);
-      const advertisingExpenseValues = months.map(month => aggregatedExpensesData[month]?.advertising_expenses || 0);
-      const entertainmentExpenseValues = months.map(month => aggregatedExpensesData[month]?.entertainment_expenses || 0);
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:99]: Lets add title comment
// Cost Of Sales

File: client/src/components/Table/TablePlanningA.tsx [hunk 13]
```diff
@@ -59,134 +91,110 @@ const TablePlanning = () => {
...
+      });
 
-      // console.log("Non Operating: " , nonOperatingIncomeValues)
-      const personnelExpensesValues = months.map(month => {
-        const totalExpenses = aggregatedExpensesData[month]?.remuneration + aggregatedPlanningAssign[month]?.assignment_unit_price +
-        aggregatedExpensesData[month]?.travel_expenses + aggregatedExpensesData[month]?.taxes_and_public_charges + 
-        aggregatedExpensesData[month]?.utilities_expenses || 0;
+      // EMPLOYEE EXPENSE
+      const totalEmployeeExpenseValues = months.map(month => {
+        const renumeration = aggregatedExpensesData[month]?.renumeration || 0;
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:127]: please make this and all relevant code to this executiveRenumeration

File: client/src/components/Table/TablePlanningA.tsx [hunk 14]
```diff
@@ -59,134 +91,110 @@ const TablePlanning = () => {
...
-      const secondHalfTotal = arr => arr.slice(6).reduce((acc, value) => acc + parseFloat(value), 0);
-      const total = arr => arr.reduce((acc, value) => acc + parseFloat(value), 0);
+      const consumableValues = months.map(month => aggregatedExpensesData[month]?.consumable_expense || 0);
+      const rentValues = months.map(month => aggregatedExpensesData[month]?.rent_expense || 0);
+      const taxesPublicChargesValues = months.map(month => aggregatedExpensesData[month]?.tax_and_public_charge || 0);
+      const depreciationExpensesValues = months.map(month => aggregatedExpensesData[month]?.depreciation_expense || 0);
+      const travelExpenseValues = months.map(month => aggregatedExpensesData[month]?.travel_expense || 0);
+      const communicationExpenseValues = months.map(month => aggregatedExpensesData[month]?.communication_expense || 0);
+      const utilitiesValues = months.map(month => aggregatedExpensesData[month]?.utilities_expense || 0);
+      const trasactionExpenseValues = months.map(month => aggregatedExpensesData[month]?.transaction_fee || 0);
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:166]: spelling mistake and misnamed - should be transactionFeesValues

Please check any other related code

File: client/src/components/Table/TablePlanningA.tsx [hunk 15]
```diff
@@ -323,56 +355,67 @@ const TablePlanning = () => {
...
           values: [
-            ...expenseTotalValues,
-            firstHalfTotal(expenseTotalValues),
-            secondHalfTotal(expenseTotalValues),
-            total(expenseTotalValues),
+            ...totalExpenseValues,
+            firstHalfTotal(totalExpenseValues),
+            secondHalfTotal(totalExpenseValues),
+            total(totalExpenseValues),
             // `${(total(expenseTotalValues) / total(costOfSalesValues) * 100).toFixed(2)}%`,
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:419]: Please Fix Comment values

File: client/src/components/Table/TablePlanningA.tsx [hunk 16]
```diff
@@ -400,17 +443,7 @@ const TablePlanning = () => {
...
-            secondHalfTotal(paymentFeeValues),
-            total(paymentFeeValues),
-            // `${(total(paymentFeeValues) / total(costOfSalesValues) * 100).toFixed(2)}%`,
-            '0',
-          ],
-        },
+        
         {
           //same " 法定福利費 "
           label: 'taxesAndPublicCharges',
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:449]: //same " 法定福利費 "

This comment is for "statutory welfare" not "taxesAndPublicCharges"

File: client/src/components/Table/TablePlanningA.tsx [hunk 17]
```diff
@@ -14,8 +14,8 @@ const TablePlanning = () => {
       window.location.href = '/login';
       return;
     }
-    // axios.get('http://127.0.0.1:8000/api/planning/all/', {
-    axios.get('http://54.178.202.58:8000/api/planning/all/', {
+    axios.get('http://127.0.0.1:8000/api/planning', {
```
[the remark this answers is not in the document]
Comment (edward-stalley-work) [reply client/src/components/Table/TablePlanningA.tsx:17]: moved to its own ticket.

File: client/src/components/Table/TablePlanningA.tsx [hunk 18]
```diff
@@ -14,8 +14,8 @@ const TablePlanning = () => {
       window.location.href = '/login';
       return;
     }
-    // axios.get('http://127.0.0.1:8000/api/planning/all/', {
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:17]: fix alignment on screen when separate projects button is pressed . (i think you already started this)

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
Thank you Ian - please see comments

File: client/src/components/Table/TablePlanningA.tsx (hunk 17 above)
[the remark this answers is not in the document]
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:17]: this is noted Ed

File: client/src/components/Table/TablePlanningA.tsx (hunk 1 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:177]: will work on it ed

File: client/src/components/Table/TablePlanningA.tsx (hunk 3 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:332]: already removed

File: client/src/components/Table/TablePlanningA.tsx (hunk 13 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:127]: already applied changes

File: client/src/components/Table/TablePlanningA.tsx (hunk 4 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:340]: removed the commented code

File: client/src/components/Table/TablePlanningA.tsx (hunk 5 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:362]: done modified

File: client/src/components/Table/TablePlanningA.tsx (hunk 6 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:373]: modified

File: client/src/components/Table/TablePlanningA.tsx (hunk 7 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:384]: modified

File: client/src/components/Table/TablePlanningA.tsx (hunk 8 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:395]: done

File: client/src/components/Table/TablePlanningA.tsx (hunk 9 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:406]: done

File: client/src/components/Table/TablePlanningA.tsx (hunk 10 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:410]: modified already added comments for each sections

File: client/src/components/Table/TablePlanningA.tsx (hunk 11 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:105]: done

File: client/src/components/Table/TablePlanningA.tsx (hunk 12 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:99]: image
already have //COST OF SALE

File: client/src/components/Table/TablePlanningA.tsx (hunk 14 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:166]: done

File: client/src/components/Table/TablePlanningA.tsx (hunk 16 above)
Comment (ian-bpoc) [reply client/src/components/Table/TablePlanningA.tsx:449]: removed the comment

File: client/src/components/Table/TablePlanningA.tsx [hunk 19]
```diff
@@ -295,85 +322,103 @@ const TablePlanning = () => {
...
-            ...renumerationValues,
-            firstHalfTotal(renumerationValues),
-            secondHalfTotal(renumerationValues),
-            total(renumerationValues),
-            // `${(total(renumerationValues) / total(costOfSalesValues) * 100).toFixed(2)}%`,
+            ...executiveRenumerationValues,
+            firstHalfTotal(executiveRenumerationValues),
+            secondHalfTotal(executiveRenumerationValues),
+            total(executiveRenumerationValues),
+            // `${(total(executiveRenumerationValues) / total(costOfSalesValues) * 100).toFixed(2)}%`,
```
Comment (edward-stalley-work) [inline client/src/components/Table/TablePlanningA.tsx:354]: total(costOfSalesValues) → total(employeeExpensesValues)

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
see comments

**Pull request merged**